config/v1/types_ingress.go

```diff
@@ -172,6 +180,98 @@ type ComponentRouteStatus struct {
 	RelatedObjects []ObjectReference `json:"relatedObjects"`
 }
 
+// RequiredRouteAnnotations specifies annotations that are required to be set on
+// newly created routes matching some criteria.
+type RequiredRouteAnnotations struct {
+	// domains is an optional list of domains for which these annotations
+	// are required.  If domains is specified and a route is created with a
+	// spec.host matching one of the domains, the route must specify the
+	// annotations specified in requiredAnnotations.  If domains is empty,
+	// the specified annotations are required for all newly created routes.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Domains string `json:"domains,omitempty"`
+
+	// policy specifies the policy for user-provided annotation values on
+	// newly created routes.  The following values are allowed for this
+	// field:
+	//
+	// * "Allow" allows the user to provide arbitrary annotation values.
+	//   With this policy, if an annotation key in requiredAnnotations is
+	//   not specified on the route at all, the route is rejected.  If every
+	//   annotation key in requiredAnnotations is specified, then the route
+	//   is admitted, irrespective of the annotation values.
+	//
+	// * "AllowWithDefaulting" allows the user to provide arbitrary
```

Comment:
Setting this as a default value on routes changes the meaning of a route depending on the cluster in which the route is created.  I think it's worth discussing with PM whether they really want to create non-replayable resources that vary based on configuration outside of the app owner's control and in some cases even outside the app owner's visibility.

I think that a cluster-admin goal of, "I want every route served for domain X to have HSTS" can be achieved without this mutation very quickly with a combination of a Deny and client-side `oc annotate route`